app/src/main/java/es/wolfi/app/passman/CredentialAdd.java

```diff
@@ -132,6 +170,107 @@ public void onDetach() {
         mListener = null;
     }
 
+    public void addSelectedFile(String encodedFile, String fileName, String mimeType, int fileSize, int requestCode) throws JSONException {
+        Context context = getContext();
+        final ProgressDialog progress = new ProgressDialog(context);
+        progress.setTitle(context.getString(R.string.loading));
+        progress.setMessage(context.getString(R.string.wait_while_loading));
+        progress.setCancelable(false);
+        progress.show();
+
+        AsyncHttpResponseHandler responseHandler = new AsyncHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, cz.msebera.android.httpclient.Header[] headers, byte[] responseBody) {
+                String result = new String(responseBody);
```

Comment:
I believe this kind of data handling would be better placed on the [api library](https://github.com/nextcloud/passman-android/tree/master/app/src/main/java/es/wolfi/passman/API)

Reply:
fixed in 5b250f8317c18764586a3d11ea130c0c944160c7